### ISSUE TITLE
Build failing on Jenkins due to sparkdemo

### ISSUE THREAD

**Issue by tswast** [GoogleCloudPlatform/java-docs-samples#85]
This might be the same flakey failure we were seeing in Travis.

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.google.appengine.sparkdemo.UserControllerTest
Feb 08, 2016 1:00:25 AM com.google.api.services.datastore.client.BaseDatastoreFactory makeClient
WARNING: Not using any credentials
Feb 08, 2016 1:00:28 AM com.google.api.services.datastore.client.BaseDatastoreFactory makeClient
WARNING: Not using any credentials
[Thread-1] INFO org.eclipse.jetty.util.log - Logging initialized @3035ms
[Thread-1] INFO spark.webserver.JettySparkServer - == Spark has ignited ...
[Thread-1] INFO spark.webserver.JettySparkServer - >> Listening on 0.0.0.0:8080
[Thread-1] INFO org.eclipse.jetty.server.Server - jetty-9.3.2.v20150730
[Thread-1] INFO org.eclipse.jetty.server.ServerConnector - Started ServerConnector@42c69b0e{HTTP/1.1,[http/1.1]}{0.0.0.0:8080}
[Thread-1] INFO org.eclipse.jetty.server.Server - Started @3198ms
[main] INFO spark.webserver.JettySparkServer - >>> Spark shutting down ...
[main] INFO org.eclipse.jetty.server.ServerConnector - Stopped ServerConnector@42c69b0e{HTTP/1.1,[http/1.1]}{0.0.0.0:8080}
[main] INFO spark.webserver.JettySparkServer - done
Tests run: 5, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.345 sec <<< FAILURE!
Running com.google.appengine.sparkdemo.UserServiceTest
Feb 08, 2016 1:00:33 AM com.google.api.services.datastore.client.BaseDatastoreFactory makeClient
WARNING: Not using any credentials
Feb 08, 2016 1:00:34 AM com.google.api.services.datastore.client.BaseDatastoreFactory makeClient
WARNING: Not using any credentials
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.119 sec

Results :

Failed tests:   testDeleteUser(com.google.appengine.sparkdemo.UserControllerTest): expected:<0> but was:<1>
```

**Comment by lesv** [comment]
@ludoch -- Any chance one of your contractors will fix this?  OR should we?

**Comment by ajkannan** [comment]
I'll start work on this now.

**Comment by tswast** [comment]
Looks like the sparkdemo target is still flaky. Jenkins build failed again last night.

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.google.appengine.sparkdemo.UserControllerTest
Feb 19, 2016 1:00:48 AM com.google.api.services.datastore.client.BaseDatastoreFactory makeClient
WARNING: Not using any credentials
Feb 19, 2016 1:00:52 AM com.google.api.services.datastore.client.BaseDatastoreFactory makeClient
WARNING: Not using any credentials
[Thread-1] INFO org.eclipse.jetty.util.log - Logging initialized @4835ms
[Thread-1] INFO spark.webserver.JettySparkServer - == Spark has ignited ...
[Thread-1] INFO spark.webserver.JettySparkServer - >> Listening on 0.0.0.0:8080
[Thread-1] INFO org.eclipse.jetty.server.Server - jetty-9.3.2.v20150730
[Thread-1] INFO org.eclipse.jetty.server.ServerConnector - Started ServerConnector@6af08186{HTTP/1.1,[http/1.1]}{0.0.0.0:8080}
[Thread-1] INFO org.eclipse.jetty.server.Server - Started @5095ms
[main] INFO spark.webserver.JettySparkServer - >>> Spark shutting down ...
[main] INFO org.eclipse.jetty.server.ServerConnector - Stopped ServerConnector@6af08186{HTTP/1.1,[http/1.1]}{0.0.0.0:8080}
[main] INFO spark.webserver.JettySparkServer - done
Tests run: 5, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 8.561 sec <<< FAILURE!
Running com.google.appengine.sparkdemo.UserServiceTest
Feb 19, 2016 1:01:03 AM com.google.api.services.datastore.client.BaseDatastoreFactory makeClient
WARNING: Not using any credentials
Feb 19, 2016 1:01:03 AM com.google.api.services.datastore.client.BaseDatastoreFactory makeClient
WARNING: Not using any credentials
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 9.296 sec

Results :

Failed tests:   testDeleteUser(com.google.appengine.sparkdemo.UserControllerTest): expected:<0> but was:<1>

Tests run: 9, Failures: 1, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] appengine-appidentity ............................. SUCCESS [13.918s]
[INFO] appengine-helloworld .............................. SUCCESS [1.267s]
[INFO] appengine-analytics ............................... SUCCESS [2.417s]
[INFO] appengine-cloudsql ................................ SUCCESS [1.852s]
[INFO] appengine-cloudstorage ............................ SUCCESS [1.990s]
[INFO] appengine-datastore ............................... SUCCESS [1.818s]
[INFO] appengine-disk .................................... SUCCESS [1.226s]
[INFO] appengine-extendingruntime ........................ SUCCESS [1.034s]
[INFO] managedvms-helloworld-mvm ......................... SUCCESS [0.795s]
[INFO] appengine-mailgun ................................. SUCCESS [1.042s]
[INFO] appengine-memcache ................................ SUCCESS [0.918s]
[INFO] appengine-sendgrid ................................ SUCCESS [1.105s]
[INFO] appengine-staticfiles ............................. SUCCESS [0.767s]
[INFO] appengine-twilio .................................. SUCCESS [1.086s]
[INFO] Example Async Rest Webapp ......................... SUCCESS [0.599s]
[INFO] sparkdemo ......................................... FAILURE [19.069s]
[INFO] doc-samples ....................................... SKIPPED
[INFO] bigquery-samples .................................. SKIPPED
[INFO] Google Cloud Datastore Snippets ................... SKIPPED
[INFO] cloud-logging-samples ............................. SKIPPED
[INFO] cloud-monitoring-samples .......................... SKIPPED
[INFO] Sample accessing the Google Cloud Storage JSON API using OAuth 2.0.  SKIPPED
[INFO] transfersample .................................... SKIPPED
[INFO] Example for the Google Cloud Storage XML API using Application Default Credentials.  SKIPPED
[INFO] Example for Google Cloud Storage using OAuth 2.0 Service Accounts on Google App Engine  SKIPPED
[INFO] defer-sample ...................................... SKIPPED
[INFO] appengine-local-testing-samples ................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 51.761s
[INFO] Finished at: Fri Feb 19 01:01:06 UTC 2016
[INFO] Final Memory: 20M/56M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.10:test (default-test) on project sparkdemo: There are test failures.
[ERROR] 
[ERROR] Please refer to /workspace/java-docs-samples/managed_vms/sparkjava/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :sparkdemo
ERROR:  (Java::Hudson::AbortException)
```

**Comment by ajkannan** [comment]
Hm, maybe it's a consistency issue.  I'll do some more digging.

**Comment by tswast** [comment]
This should be fixed with #96, which I believe clears up the issue with the tests assuming strong consistency in Datastore.
